### ISSUE TITLE
Updating suggestions prop does not update options list until next input (v6)

### ISSUE THREAD

**Issue by yrik** [i-like-robots/react-tags#209]
The options list is updated directly after the onInput custom handler https://github.com/i-like-robots/react-tags/blob/6.0/lib/ReactTags.js#L130.
However, when I query the server inside of onInput handler it takes some time.
In a result, the options list always contains list of suggestions from the previous call, not the latest one.

**Comment by i-like-robots** [comment]
Hi @yrik, data fetching is not provided by the component but implemented by yourself. Whilst I cannot debug individual implementations I'm happy to offer some advice.

Following on from the examples in #205 I'd suggest fetching more suggestions than you need to display. For example if you currently display 6 suggestion then try fetching 12 or 15 from the server. This way the `suggestionsFilter` should still have some relevant data to use until new ones are received.

**Comment by yrik** [comment]
It's a bit different problem.
Imagine I have an autocomplete for the word "**beer**"
1. user types "**be**" --> server returns [bee, beard, beetroot, beer] (we show nothing because options list is not updated as server returns data after getOptions call)
2. user types "**bee**" --> server returns [bee, beetroot, beer] we show [bee, beard, beetroot, beer], we show previously update data, not fresh one
3. user types "**beer**" -->  server returns [beer], we show  [bee, beetroot, beer], we show previously update data, not fresh one

**Comment by yrik** [comment]
In case `onInput`would be `async` it would work, of there is a way to call `const options = getOptions.call(this, query)` from custom `onInput` it would work too.

**Comment by i-like-robots** [comment]
OK I think I follow - so suggestions may be displayed that should not match the current query, is this correct? 

I'll try and recreate the issue with [this example code](https://github.com/i-like-robots/react-tags/compare/v5.13.0...async-fetch-example)

**Comment by yrik** [comment]
> OK I think I follow - so suggestions may be displayed that should not match the current query, is this correct?

Exactly! In my case it almost always shows the result of the previous query. Except for the cases when the response is cached in browser and returned instantly.

**Comment by yrik** [comment]
Also worth noticing that I'm using v6.

**Comment by i-like-robots** [comment]
After a short investigation I believe this issue has been introduced by the desire to avoid `componentWillReceiveProps()` in v6 due to changes in recent and upcoming versions of React.

It can be solved by using `getDerivedStateFromProps()` but I think this approach will require some optimisation to prevent unnecessary renders.

```js
  static getDerivedStateFromProps (props, state) {
    const options = getOptions(props, state)
    return { options }
  }
```

**Comment by i-like-robots** [comment]
Here we go, the animation below shows suggestions being updated asynchronously and the options list being updated without any other input or change:

![update](https://user-images.githubusercontent.com/271645/81659916-1502c480-9432-11ea-9644-65f56df6aca0.gif)

The solution was quite straightforward so I'll do some more testing and raise a PR.

**Comment by yrik** [comment]
I guess `useEffect` should cover this nicely as it could monitor options change.

```
useEffect(() => {
  setOptions(getOptions(props, state))
}, [options]);
```

**Comment by i-like-robots** [comment]
I have released `v6.0.0-beta.6` which should fix this issue.
